profiles/permissions.py

```diff
@@ -15,3 +15,10 @@ def has_permission(self, request, view):
         if request.query_params.get("filters") == "is_saved":
             return request.user.is_authenticated
         return True
+
+
+class UserIsAdmin(BasePermission):
+    def has_permission(self, request, view):
+        if request.method in SAFE_METHODS:
+            return True
+        return request.user.is_staff
```

Comment:
There's a DRF built-in permission [`IsAdminUser`](https://www.django-rest-framework.org/api-guide/permissions/#isadminuser). I guess it would be better if you've had something like this:
```suggestion
class IsAdminOrReadOnly(IsAdminUser):
    def has_permission(self, request, view):
        is_admin = super().has_permission(request, view)
        return request.method in SAFE_METHODS or is_admin
```

Reply:
Thank you for feedback.  I think my varient is more readable.

Reply:
Actually, the best option (and the option which I strongly recommend) will be implemenatation of narrow-focused permission classes and then combining it in `permission_classes` in view.

`Or`, `And` etc words in names of permission classes are a clear sign of violation of SRP. Thus, it's hard to reuse them etc.

So, in this case, instead of creation `IsAdminOrReadOnly` permission class, 2 permission classes (something like `IsAdminUser` + `RequestMethodIsSafe`) can be created and used as `permission_classes = (RequestMethodIsSafe | IsAdminUser)` in a view.